5_core_acc_analysis/1_stable_gene_relationships.py

```diff
@@ -425,107 +374,185 @@
+)
+expression_dist_counts_pao1_least = expression_dist_counts_pao1_least.merge(
+    pao1_least_ci[["ymin", "ymax"]], left_index=True, right_index=True
+)
+
+expression_dist_counts_pa14_most = expression_dist_counts_pa14_most.merge(
+    pa14_most_ci[["ymin", "ymax"]], left_index=True, right_index=True
+)
+expression_dist_counts_pa14_least = expression_dist_counts_pa14_least.merge(
+    pa14_least_ci[["ymin", "ymax"]], left_index=True, right_index=True
+)
+
+expression_dist_counts_pao1_all = pd.concat(
+    [expression_dist_counts_pao1_most, expression_dist_counts_pao1_least]
+)
+
+expression_dist_counts_pao1_all
+
 expression_dist_counts_pa14_all = pd.concat(
     [expression_dist_counts_pa14_most, expression_dist_counts_pa14_least]
 )
-# -
 
 expression_dist_counts_pa14_all
 
 # ### Plot
 
 # +
-# Plot PAO1 trends
-plt.figure(figsize=(10, 8))
-
-fig = sns.barplot(
-    data=expression_dist_counts_pao1_all,
-    x="offset",
-    y="normalized",
-    hue="label",
-    hue_order=[
-        "most stable acc",
-        "least stable acc",
-        "most stable core",
-        "least stable core",
-    ],
-    palette={
-        "most stable acc": "#21368B",
-        "least stable acc": "#A6AED0",
-        "most stable core": "#F8744C",
-        "least stable core": "#FCC7B7",
-    },
-    # ci=100
-)
-# TO DO:
-# Fix CI addition
-# plt.errorbar(
-#    x=['1','2','3','4','5','6','7','8','9','10','10+'],
-#    yerr=pao1_most_ci_ranges
-# )
-plt.axhline(y=1.0, color="black", linestyle="--")
-fig.legend_.remove()
-fig.set_title("Who are most/least stable core genes related to (PAO1)", fontsize=16)
-fig.set_ylabel("Odds ratio", fontsize=14)
-fig.set_xlabel("Rank correlation in expression space", fontsize=14)
-# Note: We are creating a single global legend that apply
-# to all the facets of this figure. To do this using
-# matplotlib, we need to be a little creative here
-# and add the legend to a new location that is applied
-# to the figure and then remove the legend from the facet.
-plt.legend(bbox_to_anchor=(1.05, 0.6), loc=2, borderaxespad=0.0, fontsize=12)
+pao1_subset = expression_dist_counts_pao1_all[
+    (expression_dist_counts_pao1_all["gene type"] == "acc")
+]
+pao1_subset["offset"] = list(pao1_subset["offset"].astype("str"))
+x_ticks = ["+10", "1", "2", "3", "4", "5", "6", "7", "8", "9", "10"]
+
+fig_pao1 = (
+    ggplot(pao1_subset, aes(x="offset", y="normalized", fill="label"))
+    + geom_bar(stat="identity", position="dodge", width=0.8)
+    + geom_errorbar(
+        pao1_subset,
+        aes(x="offset", ymin="ymin", ymax="ymax"),
+        position=positions.position_dodge(0.8),
+        color="darkgrey",
+    )
+    + geom_hline(aes(yintercept=1.0), linetype="dashed")
+    + labs(
+        x="Rank co-expression",
+        y=r"Fold change" + "\n" + "(% acc genes co-express/% acc genes in genome)",
+        title="Who are most/least stable core genes related to (PAO1)",
+    )
+    + theme(
```

Comment:
Could accomplish the same thing with this one liner:
```python
+ theme_seaborn("white")
```

Reply:
Nice! Thanks!